diesel/src/mysql/connection/mod.rs

```diff
@@ -12,13 +12,15 @@ use types::HasSqlType;
 
 #[allow(missing_debug_implementations, missing_copy_implementations)]
 pub struct MysqlConnection {
-    _raw_connection: RawConnection,
+    raw_connection: RawConnection,
     transaction_manager: AnsiTransactionManager,
 }
 
 impl SimpleConnection for MysqlConnection {
-    fn batch_execute(&self, _query: &str) -> QueryResult<()> {
-        unimplemented!()
+    fn batch_execute(&self, query: &str) -> QueryResult<()> {
+        self.raw_connection.enable_multi_statements(|| {
+            self.raw_connection.execute(query)
+        })
```

Comment:
In the section starting with "The following procedure outlines a suggested strategy for handling multiple statements", <https://dev.mysql.com/doc/refman/5.7/en/c-api-multiple-queries.html> talks about how you need to write a loop that calls `mysql_next_result` to actually get the results for multiple queries. Especially, it says

> For each iteration of the loop, handle the current statement result, retrieving either a result set or an affected-rows count. If an error occurs, exit the loop.

How would you interpret this? Do you maybe need to iterate the results to get all possible errors? (Currently, you are only calling `did_an_error_occur` after the initial execute and when setting the options.)